File: examples/supervision-service/app/schemas/records.py
from typing import Any

from pydantic import BaseModel, Field


class UploadResponse(BaseModel):
    """Stored upload record."""

    id: str
    original_filename: str
    stored_filename: str
    file_path: str
    file_size: int
    content_type: str | None
    created_at: str


class ProcessingJobResponse(BaseModel):
    """Processing job record."""

    id: str
    upload_id: str
    job_type: str
    status: str
    output_path: str | None
    parameters: dict[str, Any]
    error_message: str | None
    progress: int = Field(ge=0, le=100)
    current_frame: int = Field(ge=0)
    total_frames: int = Field(ge=0)
    created_at: str
    completed_at: str | None
    original_filename: str | None = None


class JobCreatedResponse(BaseModel):
    """Response after submitting an async processing job."""

    job_id: str
    upload_id: str
    status: str = "processing"


class UploadListResponse(BaseModel):
    """Paginated upload list."""

    items: list[UploadResponse]
    total: int = Field(description="Number of items returned")


class ProcessingJobListResponse(BaseModel):
    """Paginated processing job list."""

    items: list[ProcessingJobResponse]
    total: int = Field(description="Number of items returned")

File: examples/supervision-service/webapp/src/api/records.ts
export interface UploadRecord {
  id: string
  original_filename: string
  stored_filename: string
  file_path: string
  file_size: number
  content_type: string | null
  created_at: string
}

export interface ProcessingJobRecord {
  id: string
  upload_id: string
  job_type: string
  status: string
  output_path: string | null
  parameters: Record<string, unknown>
  error_message: string | null
  progress: number
  current_frame: number
  total_frames: number
  created_at: string
  completed_at: string | null
  original_filename: string | null
}

export interface UploadListResponse {
  items: UploadRecord[]
  total: number
}

export interface ProcessingJobListResponse {
  items: ProcessingJobRecord[]
  total: number
}

export async function fetchUploadRecords(limit = 50): Promise<UploadListResponse> {
  const response = await fetch(`/api/v1/records/uploads?limit=${limit}`)
  if (!response.ok) {
    throw new Error(`Failed to load upload records: ${response.status}`)
  }
  return response.json() as Promise<UploadListResponse>
}

export async function fetchJobRecords(limit = 50): Promise<ProcessingJobListResponse> {
  const response = await fetch(`/api/v1/records/jobs?limit=${limit}`)
  if (!response.ok) {
    throw new Error(`Failed to load job records: ${response.status}`)
  }
  return response.json() as Promise<ProcessingJobListResponse>
}

export function uploadFileUrl(uploadId: string): string {
  return `/api/v1/records/uploads/${uploadId}/file`
}

export function jobResultUrl(jobId: string): string {
  return `/api/v1/records/jobs/${jobId}/file`
}
